agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go

```diff
@@ -0,0 +1,664 @@
+			fv.setTerminalReason(err.Error())
+			return err
+		}
+	} else if parsedARNService == "secretsmanager" {
+		err = fv.retrieveASMCredentials(credentialsParameterARN, iamCredentials)
+		if err != nil {
+			seelog.Errorf("Failed to retrieve credentials from asm: %v", err)
+			fv.setTerminalReason(err.Error())
+			return err
+		}
+	} else {
+		err = errors.New("unsupported credentialsParameter ARN, only ssm/secretsmanager ARNs are valid")
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (fv *FSxWindowsFileServerResource) retrieveSSMCredentials(credentialsParameterARN string, iamCredentials credentials.IAMRoleCredentials) error {
+	parsedARN, err := arn.Parse(credentialsParameterARN)
+	if err != nil {
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	ssmClient := fv.ssmClientCreator.NewSSMClient(fv.region, iamCredentials)
+	ssmParam := filepath.Base(parsedARN.Resource)
+	ssmParams := []string{ssmParam}
+
+	ssmParamMap, err := ssm.GetParametersFromSSM(ssmParams, ssmClient)
+	if err != nil {
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	ssmParamData, _ := ssmParamMap[ssmParam]
+	creds := FSxWindowsFileServerCredentials{}
+
+	if err := json.Unmarshal([]byte(ssmParamData), &creds); err != nil {
+		return err
+	}
+
+	fv.SetCredentials(creds)
+	return nil
+}
+
+func (fv *FSxWindowsFileServerResource) retrieveASMCredentials(credentialsParameterARN string, iamCredentials credentials.IAMRoleCredentials) error {
+	_, err := arn.Parse(credentialsParameterARN)
+	if err != nil {
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	asmClient := fv.asmClientCreator.NewASMClient(fv.region, iamCredentials)
+	asmData, err := asm.GetSecretFromASM(credentialsParameterARN, asmClient)
+	if err != nil {
+		return err
+	}
+
+	creds := FSxWindowsFileServerCredentials{}
+	if err := json.Unmarshal([]byte(asmData), &creds); err != nil {
+		return err
+	}
+
+	fv.SetCredentials(creds)
+	return nil
+}
+
+func (fv *FSxWindowsFileServerResource) retrieveFileSystemDNSName(fileSystemId string, iamCredentials credentials.IAMRoleCredentials) error {
+	fileSystemIds := []string{fileSystemId}
+	fsxClient := fv.fsxClientCreator.NewFSxClient(fv.region, iamCredentials)
+	fileSystemDNSMap, err := fsx.GetFileSystemDNSNames(fileSystemIds, fsxClient)
+	if err != nil {
+		fv.setTerminalReason(err.Error())
+		return err
+	}
+
+	fv.SetFileSystemDNSName(fileSystemDNSMap[fileSystemId])
+	return nil
+}
+
+var execCommand = exec.Command
+
+func (fv *FSxWindowsFileServerResource) performHostMount(remotePath string, localPath string, username string, password string) error {
+	// formatting to keep powershell happy
+	localPath = fmt.Sprintf("-LocalPath %s", localPath)
+	remotePath = fmt.Sprintf("-RemotePath %s", remotePath)
+	creds := fmt.Sprintf("-Credential $(New-Object System.Management.Automation.PSCredential(%s, $(ConvertTo-SecureString %s -AsPlainText -Force)))", username, password)
+
+	cmd := execCommand("powershell.exe",
```

Comment:
could you add comments on what this command does(below one too)

Reply:
will do